contracts/pcv/utils/NamedStaticPCVDepositWrapper.sol

```diff
@@ -0,0 +1,201 @@
+pragma solidity ^0.8.4;
+
+import "../IPCVDepositBalances.sol";
+import "../../Constants.sol";
+import "../../refs/CoreRef.sol";
+import "@openzeppelin/contracts/utils/math/SafeCast.sol";
+
+/**
+  @notice a contract to report static PCV data to cover PCV not held with a reliable oracle or on-chain reading 
+  @author Fei Protocol
+
+  Returns PCV in USD terms
+*/
+contract NamedStaticPCVDepositWrapper is IPCVDepositBalances, CoreRef {
+    using SafeCast for *;
+
+    // -------------- Events ---------------
+    /// @notice event to update fei and usd balance
+    event BalanceUpdate(uint256 oldBalance, uint256 newBalance, uint256 oldFEIBalance, uint256 newFEIBalance);
+
+    /// @notice event to remove a deposit
+    event DepositRemoved(uint256 index);
+    
+    /// @notice event to add a new deposit
+    event DepositAdded(uint256 index, string indexed depositName);
+
+    /// @notice event emitted when a deposit is edited
+    event DepositChanged(uint256 index, string indexed depositName);
+
+    /// @notice struct to store info on each PCV Deposit
+    struct DepositInfo {
+        string depositName;
+        uint256 usdAmount; /// USD equivalent in this deposit, not including FEI value
+        uint256 feiAmount; /// amount of FEI in this deposit
+        uint256 underlyingTokenAmount; /// amount of underlying token in this deposit
+        address underlyingToken; /// address of the underlying token this deposit is reporting
+    }
+
+    /// @notice a list of all pcv deposits
+    DepositInfo[] public pcvDeposits;
+
+    /// @notice the PCV balance
+    uint256 public override balance;
+
+    /// @notice the reported FEI balance to track protocol controlled FEI in these deposits
+    uint256 public feiReportBalance;
+
+    constructor(address _core, DepositInfo[] memory newPCVDeposits) CoreRef(_core) {
+
+        // Uses oracle admin to share admin with CR oracle where this contract is used
+        _setContractAdminRole(keccak256("ORACLE_ADMIN_ROLE"));
+
+        // add all pcv deposits
+        for (uint256 i = 0; i < newPCVDeposits.length; i++) {
+            _addDeposit(newPCVDeposits[i]);
+        }
+    }
+
+    // ----------- Helper methods to change state -----------
+
+    /// @notice helper method to add a PCV deposit
+    function _addDeposit(DepositInfo memory newPCVDeposit) internal {
```

Comment:
Should we store a mapping of names -> DepositInfo for lookups? Can also be used to ensure that no duplicate names are added to @ditchfieldcaleb 's earlier concerns

Reply:
We shouldn't need to think adversarially about deposit names as only OA and the governor can update the list and guardian can remove deposits instantly.

If we were to do this feature, the data structure would be name -> DepositInfo Index so that we aren't storing each struct and it's data twice.